ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/forktransition/CapellaStateUpgrade.java

```diff
@@ -0,0 +1,71 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic.versions.capella.forktransition;
+
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.config.SpecConfigCapella;
+import tech.pegasys.teku.spec.datastructures.state.Fork;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.common.BeaconStateFields;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.bellatrix.BeaconStateBellatrix;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.capella.BeaconStateCapella;
+import tech.pegasys.teku.spec.logic.common.forktransition.StateUpgrade;
+import tech.pegasys.teku.spec.logic.versions.bellatrix.helpers.BeaconStateAccessorsBellatrix;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionsCapella;
+
+public class CapellaStateUpgrade implements StateUpgrade<BeaconStateCapella> {
+
+  private final SpecConfigCapella specConfig;
+  private final SchemaDefinitionsCapella schemaDefinitions;
+  private final BeaconStateAccessorsBellatrix beaconStateAccessors;
+
+  public CapellaStateUpgrade(
+      final SpecConfigCapella specConfig,
+      final SchemaDefinitionsCapella schemaDefinitions,
+      final BeaconStateAccessorsBellatrix beaconStateAccessors) {
+    this.specConfig = specConfig;
+    this.schemaDefinitions = schemaDefinitions;
+    this.beaconStateAccessors = beaconStateAccessors;
+  }
+
+  @Override
+  public BeaconStateCapella upgrade(final BeaconState preState) {
+    final UInt64 epoch = beaconStateAccessors.getCurrentEpoch(preState);
+    BeaconStateBellatrix preStateBellatrix = BeaconStateBellatrix.required(preState);
+    return schemaDefinitions
+        .getBeaconStateSchema()
+        .createEmpty()
+        .updatedCapella(
+            state -> {
+              BeaconStateFields.copyCommonFieldsFromSource(state, preState);
+
+              state.setCurrentEpochParticipation(preStateBellatrix.getCurrentEpochParticipation());
+              state.setPreviousEpochParticipation(
+                  preStateBellatrix.getPreviousEpochParticipation());
+              state.setCurrentSyncCommittee(preStateBellatrix.getCurrentSyncCommittee());
+              state.setNextSyncCommittee(preStateBellatrix.getNextSyncCommittee());
+              state.setInactivityScores(preStateBellatrix.getInactivityScores());
+
+              state.setFork(
+                  new Fork(
+                      preState.getFork().getCurrentVersion(),
+                      specConfig.getCapellaForkVersion(),
+                      epoch));
+
+              state.setLatestExecutionPayloadHeader(
+                  schemaDefinitions.getExecutionPayloadHeaderSchema().getDefault());
+              state.setLatestWithdrawalValidatorIndex(UInt64.valueOf(state.getValidators().size()));
```

Comment:
This is actually initialised to 0: https://github.com/ethereum/consensus-specs/pull/3068/files#diff-c06c6116e0fa7993a0d4d7925b64cd2e5f7a23015a93e3c4d8d51209fe9f4dc0R133

Reply:
oh so it misses. the first index, thats sad, but ok